### PR TITLE
178085341 cluster stats view

### PR DESCRIPTION
**Pull request by pawjer**
#### What's this PR do?
- Create backend part of ClusterStats page

#### How should this be manually tested?
- Go to /cluster-stats to see printed out hashes
- check and run tests

#### What are the relevant tickets?
https://www.pivotaltracker.com/story/show/178085341

#### Task completed checklist:
- [x] Is there appropriate test coverage?
- [x] Did I take into consideration possible security vulnerabilities?
- [x] Are the controllers secure?
- [x] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [Y] Does this PR add new dependencies?

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -106,13 +106,14 @@ end
 group :test do
   # Adds support for Capybara system testing and selenium driver
   gem 'capybara', '>= 2.15'
+  gem 'rails-controller-testing'
```
Comment (maciejkocylapc) [inline Gemfile:109]: add version

File: lib/core_ext/array.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    return nil if sum.nil?
+
+    sum / size.to_f
+  end
+
+  private
+
+  def guard_numbers
+    return if all? do |el|
+      el.class.in? [Numeric, Integer, BigDecimal, Float]
```
Comment (maciejkocylapc) [inline lib/core_ext/array.rb:22]: maybe el.class.superclass == Numeric ?

**Review by maciejkocylapc — COMMENTED** [review]
add tests for new methods in queries

File: Gemfile (hunk 1 above)
Comment (pawjer) [reply Gemfile:109]: For what purpose, actually?

Comment (maciejkocylapc) [reply Gemfile:109]: there may occur some dependency conflicts in future if the version changes

**Review by maciejkocylapc — APPROVED** [review]

File: lib/core_ext/array.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Array class extensions
+class Array
```
Comment (radek-polcode) [inline lib/core_ext/array.rb:4]: Write tests for those methods.

File: Gemfile.lock [hunk 4]
```diff
@@ -436,4 +441,4 @@ DEPENDENCIES
   whenever
 
 BUNDLED WITH
-   2.1.4
+   2.2.16
```
[the remark this answers is not in the document]
Comment (pawjer) [reply Gemfile.lock:444]: Good catch!

File: lib/core_ext/array.rb (hunk 3 above)
Comment (pawjer) [reply lib/core_ext/array.rb:4]: Done!

**Review by radek-polcode — APPROVED** [review]

**Pull request merged**